OsmAndMapCreator/src/net/osmand/BinaryMerger.java

```diff
@@ -286,6 +302,71 @@ private void combineAddressIndex(String name, BinaryMapIndexWriter writer, Addre
 		writer.endWriteAddressIndex();
 	}
 
+	private void combinePoiIndex(String name, BinaryMapIndexWriter writer, long dateCreated, PoiRegion[] poiRegions, BinaryMapIndexReader[] indexes)
+			throws IOException, SQLException {
+		int readPoiCount = 0;
+		int badNeg = 0;
+//		int writtenPoiCount = 0;
+		MapRenderingTypesEncoder renderingTypes = new MapRenderingTypesEncoder(null, name);
+		boolean overwriteIds = true;
+		IndexPoiCreator indexPoiCreator = new IndexPoiCreator(renderingTypes, overwriteIds);
+		indexPoiCreator.createDatabaseStructure(new File(new File(System.getProperty("user.dir")), IndexCreator.getPoiFileName(name)));
+		List<Long> ids = new ArrayList<Long>();
```

Comment:
Why this List is needed at all? It takes huge memory, even there is a need it should be TLongHashSet used in set.